webapp/components/Embeds/VideoEmbed.vue

```diff
@@ -0,0 +1,110 @@
+<template>
+  <div class="video-embed">
+    <div class="video-embed-wrapper" v-if="embedUrl">
+      <div v-if="!renderEmbed" class="video-embed-preview" @click="renderEmbed = true">
+        <img :src="embedData.image" />
+        <div class="player-icon">
+          <ds-icon name="youtube-play" size="xxx-large" />
+        </div>
+      </div>
+      <iframe
+        v-else
+        :src="url"
+        frameborder="0"
+        webkitallowfullscreen
+        mozallowfullscreen
+        allowfullscreen
+      ></iframe>
+    </div>
+    <default-embed v-else :previewImage="embedData.image" :url="url" />
+  </div>
+</template>
+
+<script>
+import DefaultEmbed from './DefaultEmbed.vue'
+
+export default {
+  name: 'video-embed',
+  components: {
+    DefaultEmbed,
+  },
+  props: {
+    url: {
+      type: String,
+      default: '',
+    },
+    embedData: {
+      type: Object,
+      default() {
+        return {}
+      },
+    },
+  },
+  data() {
+    return {
+      renderEmbed: false,
+    }
+  },
+  computed: {
+    embedUrl() {
+      if (this.embedData.publisher === 'YouTube') {
```

Comment:
No, I don't want to see this kind of complexity. I would prefer to stick to the current solution for more complex embeds. 

 For security, have a look if we can trust the oembed provider. For data privacy, wrap the embed with an additional click (are you OK to load stuff from 3rd parties?). For mobile responsiveness, try out if the iframe or whatever breaks the layout.